test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -94,9 +127,35 @@ public static void main(String[] args) throws IOException {
     }
 
     static void sendDatagram(DatagramChannel dc, SocketAddress remote)
-        throws IOException
-    {
-        ByteBuffer bb = ByteBuffer.wrap("Greetings!".getBytes());
+            throws IOException {
+        ByteBuffer bb = ByteBuffer.wrap(GREETINGS_MESSAGE.getBytes());
         dc.send(bb, remote);
     }
+
+    static boolean checkUnexpectedWakeup(Set<SelectionKey> selectedKeys) {
+        System.out.format("Received %d keys%n", selectedKeys.size());
+        for (SelectionKey key : selectedKeys) {
+            if (!key.isValid() || !key.isReadable()) {
+                System.out.println("Invalid or unreadable key: " + key);
+                continue;
+            }
+            try {
+                System.out.println("Attempting to read datagram from key: " + key);
+                DatagramChannel datagramChannel = (DatagramChannel) key.channel();
+                ByteBuffer buf = ByteBuffer.allocate(100);
+                SocketAddress sa = datagramChannel.receive(buf);
+                String message = new String(buf.array());
+                if (sa != null) {
+                    System.out.format("received %s at %s from %s%n", message, datagramChannel.getLocalAddress(), sa);
+                }
+                //If any received data contains the message from sendDatagram then return true
+                if (message.contains(GREETINGS_MESSAGE)) {
+                    return true;
+                }
```

Comment:
That's wrong. Remember: we want to ignore stray wake-ups, and a stray wake-ups are defined as receiving a datagram that doesn't come from our own test.